src/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -2267,7 +2267,15 @@ internal void AdjustScale(int digits, bool fRound)
                 throw new OverflowException(SQLResource.ArithOverflowMessage);
 
             bNewScale = (byte)(lAdjust + _bScale);
-            bNewPrec = (byte)(Math.Min(s_NUMERIC_MAX_PRECISION, Math.Max(1, lAdjust + _bPrec)));
+
+            if (Value == Decimal.Zero && _bScale == 0 && lAdjust > 0)
+            {
+                bNewPrec = bNewScale;
```

Comment:
Seems like it would be more clear to just reduce the adjustment being made to precision by 1 in this case, rather than just setting it to the scale.